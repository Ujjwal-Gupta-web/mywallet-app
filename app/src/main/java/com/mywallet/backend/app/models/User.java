package com.mywallet.backend.app.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

@Document(collection = "user")
public class User {
    @Id
    @Indexed(unique = true)
    @Field("username")
    private String username;
    private String password;

    @JsonIgnore
    private boolean isAdmin;

    public User(String username, String password) {
        this.username = username;
        this.password = password;
        this.isAdmin = false;
    }

    @Override
    public String toString() {
        return "User{" +
                "username='" + username + '\'' +
                ", password='" + password + '\'' +
                ", isAdmin=" + isAdmin +
                '}';
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public boolean isAdmin() {
        return isAdmin;
    }

}
