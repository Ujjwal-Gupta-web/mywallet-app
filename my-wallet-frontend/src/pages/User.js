import { Button } from "flowbite-react"
import { useEffect, useState } from "react"
import { HiArrowLeft, HiBackspace } from "react-icons/hi"
import { useDispatch, useSelector } from "react-redux"
import { Link } from "react-router-dom"
import { convertToNumberString } from "../utility/convertToNumberString"
import { getAccountStatementAction } from "../store/actions"

const UserHome = ({ children }) => {
    const balance=useSelector(state=>state.user.balance)
    const [showBalance, setShowBalance] = useState(false);
    const dispatch=useDispatch();
    useEffect(()=>{
        dispatch(getAccountStatementAction())
    },[dispatch])
    return (
        <div>
            <div className='flex items-center justify-between'>
                <div>
                    <Link to="/">
                        <Button color='gray'><span className='mr-3'><HiArrowLeft /></span>Back </Button>
                    </Link>
                </div>
                <div>
                    <Button color='gray' onClick={() => setShowBalance(!showBalance)}>
                        {showBalance ? <>Available Balance : â‚¹{convertToNumberString(balance)}</> : <>Show Balance</>}
                    </Button>
                </div>
            </div>
            <div className="m-5">
                {children}
            </div>
        </div>
    )
}

export default UserHome



