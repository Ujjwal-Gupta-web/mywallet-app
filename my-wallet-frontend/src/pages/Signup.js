import { Button, Card, Label, TextInput } from 'flowbite-react';
import { useState } from 'react';
import { HiArrowCircleLeft, HiArrowCircleRight, HiArrowRight, HiCheck, HiLockClosed, HiMail } from 'react-icons/hi';
import User from "../controllers/user"
import { Email } from "../controllers/email"
import toast from 'react-hot-toast';

const Signup = () => {
    const [user, setUser] = useState({ username: "", password: "" });
    const [isClicked, setIsClicked] = useState(false);
    const [otp, setOtp] = useState("")

    const getOTP = async () => {
        setIsClicked(true)
        const toastId = toast.loading("Please Wait....")
        const res = await Email.sendOtp(user.username);
        if (res.tag) {
            toast.success(res.message, { id: toastId });
            localStorage.setItem("OTP", res.data);
        }
        else toast.error(res.message, { id: toastId })
        setIsClicked(false)
    }

    const verifyOTP = (e) => {
        setIsClicked(true)
        const toastId = toast.loading("Please Wait....")
        const actualOtp = localStorage.getItem("OTP");
        if (actualOtp === otp) {
            toast.success("OTP verification success, please proceed immediately");
            localStorage.removeItem("OTP")
        }
        else toast.error("OTP not valid")
        setIsClicked(false)
    }

    const handleSignup = async () => {
        setIsClicked(true)
        const toastId = toast.loading("Please Wait....")
        const res = await User.signup(user);
        if (res.tag) {
            toast.success(res.message, { id: toastId });
            window.location.href = "/login"
        }
        else toast.error(res.message, { id: toastId })
        setIsClicked(true)
    }

    const handleChange = (e) => {
        setUser({ ...user, [e.target.name]: e.target.value });
    }

    return (<div>

        <Card className="max-w-sm mx-auto">

            <h3 className='text-xl'>Signup</h3>

            {/* USERNAME */}
            <>
                <div className="max-w-md ">
                    <div className="mb-2 block">
                        <Label htmlFor="username" value="Your email (username)" />
                    </div>
                    <TextInput id="username" type="email" rightIcon={HiMail} placeholder="name@flowbite.com" required
                        name="username"
                        value={user.username}
                        onChange={handleChange}
                    />
                </div>
                <Button color='gray'>Get OTP <span className='mx-3'
                    onClick={getOTP}
                    disabled={isClicked}
                ><HiArrowRight /></span></Button>
            </>

            {/* VERIFY OTP */}
            <>
                <div className="max-w-md">
                    <div className="mb-2 block">
                        <Label htmlFor="otp" value="OTP" />
                    </div>
                    <TextInput id="otp" type="tel" required
                        value={otp}
                        onChange={(e) => setOtp(e.target.value)}
                    />
                </div>
                <Button color='gray'
                    onClick={verifyOTP}
                    disabled={isClicked}
                >Verify OTP <span className='mx-3'><HiArrowRight /></span></Button>
            </>


            {/* ENTER PASSWORD */}
            <>
                <div className="max-w-md ">
                    <div className="mb-2 block">
                        <Label htmlFor="password" value="Your password" />
                    </div>
                    <TextInput id="password" type="password" rightIcon={HiLockClosed} placeholder="" required
                        name='password'
                        value={user.password}
                        onChange={handleChange}
                    />
                </div>
                <Button color='gray'
                    onClick={handleSignup}
                    disabled={isClicked}
                >Signup<span className='mx-3'><HiCheck /></span></Button>
            </>


        </Card >
    </div>

    );
}

export default Signup
