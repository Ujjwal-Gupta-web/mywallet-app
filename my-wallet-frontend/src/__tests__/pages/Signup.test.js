import React from 'react';
import { render, fireEvent, waitFor, getByTestId } from '@testing-library/react';
import Signup from '../../pages/Signup';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import User from '../../controllers/user';
import Email from '../../controllers/email';
import { act } from 'react-dom/test-utils';


const mockStore = configureStore([]);
jest.mock('../../controllers/user', () => ({
    signup: jest.fn(),
}));

jest.mock('../../controllers/email', () => ({
    sendOtp: jest.fn(),
}));

describe('Signup component', () => {
    let store;

    beforeEach(() => {
        store = mockStore({
            user: {
                isAuth: false,
            },
        });
    });

    it('snapshot testing for signup', () => {
        const container = render(<Provider store={store}>
            <Router>
                <Signup />
            </Router>
        </Provider>);

        expect(container).toMatchSnapshot();
    });

    it('should render Signup component properly', () => {
        const { getByText, getAllByText, getByLabelText } = render(<Provider store={store}>
            <Router>
                <Signup />
            </Router>
        </Provider>);

        expect(getAllByText("Signup").length).toBe(2);
        expect(getByLabelText('Your email (username)')).toBeInTheDocument();
        expect(getByLabelText('OTP')).toBeInTheDocument();
        expect(getByLabelText('Your password')).toBeInTheDocument();
    });

    it('Singup flow', async() => {
        const { getByText, getAllByText,getByRole, getByLabelText } = render(<Provider store={store}>
            <Router>
                <Signup />
            </Router>
        </Provider>);

        const emailInput = getByLabelText('Your email (username)');
        const otpInput = getByLabelText('OTP');
        const passwordInput = getByLabelText('Your password');

        fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
        fireEvent.change(otpInput, { target: { value: '123456' } });
        fireEvent.change(passwordInput, { target: { value: 'password123' } });

        expect(emailInput.value).toBe('test@example.com');
        expect(otpInput.value).toBe('123456');
        expect(passwordInput.value).toBe('password123');

        const sendOtpBtn=getByText("Get OTP");
        const verifyOtpBtn=getByText("Verify OTP");
        const signupBtn=getByRole('button', { name: /Signup/i });

        // await act(async () => {
        //     fireEvent.click(sendOtpBtn);
        //     await waitFor(() => {
        //         expect(User.signup()).toHaveBeenCalledTimes(1);
        //     });
        // });

    });

});
