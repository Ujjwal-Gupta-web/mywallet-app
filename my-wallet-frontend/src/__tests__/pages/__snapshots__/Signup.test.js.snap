// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component snapshot testing for signup 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flex rounded-lg border border-gray-200 bg-white shadow-md dark:border-gray-700 dark:bg-gray-800 flex-col max-w-sm mx-auto"
          data-testid="flowbite-card"
        >
          <div
            class="flex h-full flex-col justify-center gap-4 p-6"
          >
            <h3
              class="text-xl"
            >
              Signup
            </h3>
            <div
              class="max-w-md "
            >
              <div
                class="mb-2 block"
              >
                <label
                  class="text-sm font-medium text-gray-900 dark:text-white"
                  data-testid="flowbite-label"
                  for="username"
                >
                  Your email (username)
                </label>
              </div>
              <div
                class="flex"
              >
                <div
                  class="relative w-full"
                >
                  <div
                    class="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-3"
                    data-testid="right-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-gray-500 dark:text-gray-400"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 20 20"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M2.003 5.884L10 9.882l7.997-3.998A2 2 0 0016 4H4a2 2 0 00-1.997 1.884z"
                      />
                      <path
                        d="M18 8.118l-8 4-8-4V14a2 2 0 002 2h12a2 2 0 002-2V8.118z"
                      />
                    </svg>
                  </div>
                  <input
                    class="block w-full border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 p-2.5 text-sm pr-10 rounded-lg"
                    id="username"
                    name="username"
                    placeholder="name@flowbite.com"
                    required=""
                    type="email"
                    value=""
                  />
                </div>
              </div>
            </div>
            <button
              class="group flex items-center justify-center p-0.5 text-center font-medium relative focus:z-10 focus:outline-none text-gray-900 bg-white border border-gray-200 enabled:hover:bg-gray-100 enabled:hover:text-cyan-700 :ring-cyan-700 focus:text-cyan-700 dark:bg-transparent dark:text-gray-400 dark:border-gray-600 dark:enabled:hover:text-white dark:enabled:hover:bg-gray-700 rounded-lg focus:ring-2"
              type="button"
            >
              <span
                class="flex items-center transition-all duration-200 rounded-md text-sm px-4 py-2"
              >
                Get OTP 
                <span
                  class="mx-3"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <div
              class="max-w-md"
            >
              <div
                class="mb-2 block"
              >
                <label
                  class="text-sm font-medium text-gray-900 dark:text-white"
                  data-testid="flowbite-label"
                  for="otp"
                >
                  OTP
                </label>
              </div>
              <div
                class="flex"
              >
                <div
                  class="relative w-full"
                >
                  <input
                    class="block w-full border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 p-2.5 text-sm rounded-lg"
                    id="otp"
                    required=""
                    type="tel"
                    value=""
                  />
                </div>
              </div>
            </div>
            <button
              class="group flex items-center justify-center p-0.5 text-center font-medium relative focus:z-10 focus:outline-none text-gray-900 bg-white border border-gray-200 enabled:hover:bg-gray-100 enabled:hover:text-cyan-700 :ring-cyan-700 focus:text-cyan-700 dark:bg-transparent dark:text-gray-400 dark:border-gray-600 dark:enabled:hover:text-white dark:enabled:hover:bg-gray-700 rounded-lg focus:ring-2"
              type="button"
            >
              <span
                class="flex items-center transition-all duration-200 rounded-md text-sm px-4 py-2"
              >
                Verify OTP 
                <span
                  class="mx-3"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <div
              class="max-w-md "
            >
              <div
                class="mb-2 block"
              >
                <label
                  class="text-sm font-medium text-gray-900 dark:text-white"
                  data-testid="flowbite-label"
                  for="password"
                >
                  Your password
                </label>
              </div>
              <div
                class="flex"
              >
                <div
                  class="relative w-full"
                >
                  <div
                    class="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-3"
                    data-testid="right-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-gray-500 dark:text-gray-400"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 20 20"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M5 9V7a5 5 0 0110 0v2a2 2 0 012 2v5a2 2 0 01-2 2H5a2 2 0 01-2-2v-5a2 2 0 012-2zm8-2v2H7V7a3 3 0 016 0z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                  <input
                    class="block w-full border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 p-2.5 text-sm pr-10 rounded-lg"
                    id="password"
                    name="password"
                    placeholder=""
                    required=""
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
            <button
              class="group flex items-center justify-center p-0.5 text-center font-medium relative focus:z-10 focus:outline-none text-gray-900 bg-white border border-gray-200 enabled:hover:bg-gray-100 enabled:hover:text-cyan-700 :ring-cyan-700 focus:text-cyan-700 dark:bg-transparent dark:text-gray-400 dark:border-gray-600 dark:enabled:hover:text-white dark:enabled:hover:bg-gray-700 rounded-lg focus:ring-2"
              type="button"
            >
              <span
                class="flex items-center transition-all duration-200 rounded-md text-sm px-4 py-2"
              >
                Signup
                <span
                  class="mx-3"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flex rounded-lg border border-gray-200 bg-white shadow-md dark:border-gray-700 dark:bg-gray-800 flex-col max-w-sm mx-auto"
        data-testid="flowbite-card"
      >
        <div
          class="flex h-full flex-col justify-center gap-4 p-6"
        >
          <h3
            class="text-xl"
          >
            Signup
          </h3>
          <div
            class="max-w-md "
          >
            <div
              class="mb-2 block"
            >
              <label
                class="text-sm font-medium text-gray-900 dark:text-white"
                data-testid="flowbite-label"
                for="username"
              >
                Your email (username)
              </label>
            </div>
            <div
              class="flex"
            >
              <div
                class="relative w-full"
              >
                <div
                  class="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-3"
                  data-testid="right-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="h-5 w-5 text-gray-500 dark:text-gray-400"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.003 5.884L10 9.882l7.997-3.998A2 2 0 0016 4H4a2 2 0 00-1.997 1.884z"
                    />
                    <path
                      d="M18 8.118l-8 4-8-4V14a2 2 0 002 2h12a2 2 0 002-2V8.118z"
                    />
                  </svg>
                </div>
                <input
                  class="block w-full border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 p-2.5 text-sm pr-10 rounded-lg"
                  id="username"
                  name="username"
                  placeholder="name@flowbite.com"
                  required=""
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
          <button
            class="group flex items-center justify-center p-0.5 text-center font-medium relative focus:z-10 focus:outline-none text-gray-900 bg-white border border-gray-200 enabled:hover:bg-gray-100 enabled:hover:text-cyan-700 :ring-cyan-700 focus:text-cyan-700 dark:bg-transparent dark:text-gray-400 dark:border-gray-600 dark:enabled:hover:text-white dark:enabled:hover:bg-gray-700 rounded-lg focus:ring-2"
            type="button"
          >
            <span
              class="flex items-center transition-all duration-200 rounded-md text-sm px-4 py-2"
            >
              Get OTP 
              <span
                class="mx-3"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </button>
          <div
            class="max-w-md"
          >
            <div
              class="mb-2 block"
            >
              <label
                class="text-sm font-medium text-gray-900 dark:text-white"
                data-testid="flowbite-label"
                for="otp"
              >
                OTP
              </label>
            </div>
            <div
              class="flex"
            >
              <div
                class="relative w-full"
              >
                <input
                  class="block w-full border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 p-2.5 text-sm rounded-lg"
                  id="otp"
                  required=""
                  type="tel"
                  value=""
                />
              </div>
            </div>
          </div>
          <button
            class="group flex items-center justify-center p-0.5 text-center font-medium relative focus:z-10 focus:outline-none text-gray-900 bg-white border border-gray-200 enabled:hover:bg-gray-100 enabled:hover:text-cyan-700 :ring-cyan-700 focus:text-cyan-700 dark:bg-transparent dark:text-gray-400 dark:border-gray-600 dark:enabled:hover:text-white dark:enabled:hover:bg-gray-700 rounded-lg focus:ring-2"
            type="button"
          >
            <span
              class="flex items-center transition-all duration-200 rounded-md text-sm px-4 py-2"
            >
              Verify OTP 
              <span
                class="mx-3"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </button>
          <div
            class="max-w-md "
          >
            <div
              class="mb-2 block"
            >
              <label
                class="text-sm font-medium text-gray-900 dark:text-white"
                data-testid="flowbite-label"
                for="password"
              >
                Your password
              </label>
            </div>
            <div
              class="flex"
            >
              <div
                class="relative w-full"
              >
                <div
                  class="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-3"
                  data-testid="right-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="h-5 w-5 text-gray-500 dark:text-gray-400"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M5 9V7a5 5 0 0110 0v2a2 2 0 012 2v5a2 2 0 01-2 2H5a2 2 0 01-2-2v-5a2 2 0 012-2zm8-2v2H7V7a3 3 0 016 0z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
                <input
                  class="block w-full border disabled:cursor-not-allowed disabled:opacity-50 bg-gray-50 border-gray-300 text-gray-900 focus:border-cyan-500 focus:ring-cyan-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-cyan-500 dark:focus:ring-cyan-500 p-2.5 text-sm pr-10 rounded-lg"
                  id="password"
                  name="password"
                  placeholder=""
                  required=""
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <button
            class="group flex items-center justify-center p-0.5 text-center font-medium relative focus:z-10 focus:outline-none text-gray-900 bg-white border border-gray-200 enabled:hover:bg-gray-100 enabled:hover:text-cyan-700 :ring-cyan-700 focus:text-cyan-700 dark:bg-transparent dark:text-gray-400 dark:border-gray-600 dark:enabled:hover:text-white dark:enabled:hover:bg-gray-700 rounded-lg focus:ring-2"
            type="button"
          >
            <span
              class="flex items-center transition-all duration-200 rounded-md text-sm px-4 py-2"
            >
              Signup
              <span
                class="mx-3"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
